nix/haskell.nix

```diff
@@ -1,52 +1,277 @@
-{ sourcesFile ? ./sources.json
-, system ? builtins.currentSystem
-, sources ? import ./sources.nix { inherit system sourcesFile; }
-, plutus ? import sources.plutus { }
-, plutus-apps ? import sources.plutus-apps { }
-, deferPluginErrors ? true
-, doCoverage ? false }:
-let inherit (plutus-apps) pkgs;
-    utils = import ./utils.nix;
-in pkgs.haskell-nix.cabalProject rec {
-  src = pkgs.haskell-nix.haskellLib.cleanGit {
-    name = "gero-gov";
-    src = ./..;
-  };
+{ src, inputs, pkgs, doCoverage ? false, deferPluginErrors ? true, ... }:
 
-  # Plutus uses a patched GHC. And so shall we.
-  compiler-nix-name = "ghc810420210212";
+pkgs.haskell-nix.cabalProject {
+  inherit src;
 
-  # -- Materialization
-  # See https://input-output-hk.github.io/haskell.nix/tutorials/materialization/:
-  # Update using:
-  #   nix-build default.nix 2>&1 | grep -om1 '/nix/store/.*-updateMaterialized' | bash
-  # plan-sha256 = "0m56bhk9w3v1zqpig84f9krrp6sqg21w0vxbjiqcxz8n7c39aw54";
-  # materialized = ./materialization/gero-gov.materialized;
+  name = "plutus-simple-model";
 
-  modules = [{
-    packages = {
-      cardano-crypto-praos.components.library.pkgconfig =
-        pkgs.lib.mkForce [ [ pkgs.libsodium-vrf ] ];
+  compiler-nix-name = "ghc8107";
 
-      cardano-crypto-class.components.library.pkgconfig =
-        pkgs.lib.mkForce [ [ pkgs.libsodium-vrf ] ];
+  shell = {
+    inputsFrom = [ pkgs.libsodium-vrf ];
+
+    # Make sure to keep this list updated after upgrading git dependencies!
+    additional = ps:
+      with ps; [
+        filemanip
+        ieee
+        base-deriving-via
+        cardano-addresses
+        cardano-addresses-cli
+        cardano-binary
+        cardano-crypto
+        cardano-crypto-class
+        cardano-crypto-praos
+        cardano-crypto-wrapper
+        cardano-ledger-alonzo
+        cardano-ledger-byron
+        cardano-ledger-core
+        cardano-ledger-pretty
+        cardano-ledger-shelley
+        cardano-ledger-shelley-ma
+        cardano-prelude
+        cardano-slotting
+        flat
+        freer-extras
+        goblins
+        measures
+        orphans-deriving-via
+        playground-common
+        plutus-chain-index
+        plutus-ledger-constraints
+        plutus-contract
+        plutus-core
+        plutus-ledger
+        plutus-ledger-api
+        plutus-pab
+        plutus-playground-server
+        plutus-tx
+        plutus-tx-plugin
+        plutus-use-cases
+        prettyprinter-configurable
+        quickcheck-dynamic
+        Win32-network
+        word-array
+      ];
+
+    withHoogle = true;
+
+    tools = {
+      cabal = "latest";
+      haskell-language-server = "latest";
+    };
+
+    exactDeps = true;
 
-      marlowe.doHaddock = deferPluginErrors;
-      marlowe.flags.defer-plugin-errors = deferPluginErrors;
+    nativeBuildInputs = with pkgs;
```

Comment:
You want to use `pkgs` without any overlays, and use `cabal-install` from that `pkgs`. Otherwise you'll have to build it yourself.